07-routing-gossip.md

```diff
@@ -118,84 +118,93 @@ announcement message: this is accomplished by having a signature from each
-
-The receiving node MUST ignore the message if the output specified
-by `short_channel_id` does not
-correspond to a P2WSH using `bitcoin_key_1` and `bitcoin_key_2` as
-specified in [BOLT #3](03-transactions.md#funding-transaction-output).
-The receiving node MUST ignore the message if this output is spent.
-
-The receiving node MUST ignore the message if the specified `chain_hash`
-is unknown to the receiver.
-
-Otherwise, the receiving node SHOULD fail the connection if
-`bitcoin_signature_1`, `bitcoin_signature_2`, `node_signature_1` or
-`node_signature_2` are invalid or not correct.
-
-Otherwise, if `node_id_1` or `node_id_2` are blacklisted, it SHOULD
-ignore the message.
-
-Otherwise, if the transaction referred to was not previously announced
-as a channel, the receiving node SHOULD queue the message for
-rebroadcasting, but MAY choose not to for messages longer than
-the minimum expected length. If it has previously received a valid
-`channel_announcement` for the same transaction in the same block, but
-for a different `node_id_1` or `node_id_2`, it SHOULD blacklist the
-previous message's `node_id_1` and `node_id_2` as well as this
-`node_id_1` and `node_id_2` and forget channels connected to them,
-otherwise it SHOULD store this `channel_announcement`.
-
-The receiving node SHOULD forget a channel once its funding output has
-been spent or reorganized out.
+The creating node:
+  - MUST set `chain_hash` to the 32-byte hash that uniquely identifies the chain
+  that the channel was opened within:
+    - MUST set the _Bitcoin blockchain_ `chain_hash` value (encoded in hex) to: `000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f`.
+  - MUST set `short_channel_id` to refer to the confirmed funding transaction,
+  as specified in [BOLT #2](02-peer-protocol.md#the-funding_locked-message).
+    - Note: the corresponding output MUST be a P2WSH, as described in [BOLT #3](03-transactions.md#funding-transaction-output).
+  - MUST set `node_id_1` and `node_id_2` to the public keys of the two nodes
+  operating the channel, such that `node_id_1` is the numerically-lesser of the
+  two DER-encoded keys sorted in ascending numerical order.
+  - MUST set `bitcoin_key_1` and `bitcoin_key_2` to `node_id_1` and `node_id_2`'s
+  respective `funding_pubkey`s.
+  - MUST compute the double-SHA256 hash `h` of the message, beginning at offset
+  256, up to the end of the message.
+    - Note: the hash skips the 4 signatures but hashes the rest of the message,
+    including any future fields appended to the end.
+  - MUST verify that `node_signature_1` and `node_signature_2` are valid
+    signatures of the hash `h`, using `node_id_1` and `node_id_2`'s respective
+    secrets.
+  - MUST verify that `bitcoin_signature_1` and `bitcoin_signature_2` are valid
+  signatures of the hash `h`, using `bitcoin_key_1` and `bitcoin_key_2`'s
+  respective secrets.
```

Comment:
Instead of "MUST verify" this should be "MUST set", same with the node_signature above. One of them will be the one the node generates itself, the other will be the one the peer sends in the `announcement_signatures` message.

Reply:
fixed